pkg/client/cli/intercept/docker_run.go

```diff
@@ -0,0 +1,218 @@
+package intercept
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"os/signal"
+	"strings"
+	"time"
+
+	"github.com/datawire/dlib/dcontext"
+	"github.com/datawire/dlib/dexec"
+	"github.com/datawire/dlib/dlog"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/cli/flags"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/cli/output"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/docker"
+	"github.com/telepresenceio/telepresence/v2/pkg/errcat"
+	"github.com/telepresenceio/telepresence/v2/pkg/proc"
+)
+
+func (s *state) prepareDockerRun(ctx context.Context) error {
+	imageName, idx := firstDockerArg(s.Cmdline)
+	// Ensure that the image is ready to run before we create the intercept.
+	if s.DockerBuild == "" {
+		if idx < 0 {
+			return errcat.User.New(`unable to find the image name. When using --docker-run, the syntax after "--" must be [OPTIONS] IMAGE [COMMAND] [ARG...]`)
+		}
+		return docker.PullImage(ctx, imageName)
+	}
+
+	// DockerBuild will produce an image-ID that must be injected into the docker run argument
+	// list. It must be injected between arguments intended for docker run and arguments intended
+	// for the container, so we require that a placeholder is present. E.g.
+	//
+	// telepresence intercept hello --docker-build ./some/path -- -it --add-host foo IMAGE --port 8080
+	if (idx < 0 || imageName != "IMAGE") && len(s.Cmdline) > 0 {
+		return errcat.User.New(`` +
```

Comment:
Empty `` ?

Reply:
Yeah, because otherwise the formatting of the lines that follow gets messed up.